zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -438,6 +461,10 @@ impl UpdateWith<Option<transaction::JoinSplitData<Groth16Proof>>> for Chain {
         joinsplit_data: &Option<transaction::JoinSplitData<Groth16Proof>>,
     ) -> Result<(), ValidateContextError> {
         if let Some(joinsplit_data) = joinsplit_data {
+            for cm in joinsplit_data.note_commitments() {
+                self.sprout_note_commitment_tree.append(cm);
+            }
```

Comment:
Missing docs: let's explain why we can't remove the sprout notes in `revert_chain_state_with`

Reply:
Added in dce91134

Reply:
I had another thought here - can we assert that the note commitment tree is empty instead?
Or is it a copy of the tree from the finalized state at this point?

It's not worth spending a lot of time on. I just wondered if we could do something stronger than a comment.

Reply:
That makes sense, maybe we should split the tip and root methods in a future refactor.
But as long as we have tests I think we're good here.